### DISCUSSION TITLE
Delete a table and custom query when data source is Mysql

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#11232] [Help] [answered]
Is there a way to remove a table that we don't need to be synced or has been deleted from Mysql db from a synced Mysql datasource?
Same question for custom mysql queries implemented in Budibase. Can we delete somehow a query we don't need? 
I don't see a delete option at Budibase admin backend.

Version Self host: 2.5.0

**Comment by ConorWebb96** [comment]
Hey @gmentzik,

You don't have to delete tables from the source to remove them. All you need to do is refetch them and deselect the ones you don't want added to your Budibase base instance. I've attached a gif below showing this;
![Refetch external dbs](https://github.com/Budibase/budibase/assets/126772285/4bb88407-b3db-4aa7-9f8c-2ae5064fda06)

This should work with any all the query based data sources e.g. MySQL, MsSQL, postgres etc. I believe the version require is 2.8.2 and up.

As for your second question you can also delete queries, simply hover over the query, click the ellipsis `...` and then click on the delete text. This should delete the query you are actively hovering over.
<img width="389" alt="Screenshot 2023-07-14 at 08 22 52" src="https://github.com/Budibase/budibase/assets/126772285/4a12cd41-12b8-4c28-a9f7-dec59537250d">


I hope this helps.

**Reply by gmentzik** [reply]
I am using Budibase 2.5. I saw that at 2.8 Fetching behavior has changed. But before updating to it I need to now if the bug that was covered by workaround described at User Guide is now fixed and from which version?

https://docs.budibase.com/docs/sql-datasource#fetching-specific-tables
Another reason may be to prevent changes to certain tables being overwritten in Budibase; particularly [Formula](https://docs.budibase.com/docs/formula) columns.

See also question: https://github.com/Budibase/budibase/discussions/11182

**Reply by ConorWebb96** [reply, marked as answer]
Hey @gmentzik,

I've found the issue that was created around this it seems to have been fixed: https://github.com/Budibase/budibase/issues/10193. Additionally, when testing and refetching tables that had relationships/formulas it didn't seem to do anything (the formula and relationship stayed). 

The only way you will loose a formula/relationship is when you deselect a table and remove it from being fetch, deselecting an refetching deletes all references of that table on Budibase. This means anything added will be remove and thus needs re-added.

The above is intended behaviour, if this is not the way you wish for it to work I'd recommend creating a feature request and explain in detail how you wish it to work instead.

The docs have also been updated, with new images so thanks for pointing that out! 🚀 

I'd recommend upgrading to the to at least 2.8 +

I hope this makes sense and is helpful.

**Reply by gmentzik** [reply]
Thank you for the info. I will test 2.8 it in a test environment again to check behavior. 
I believe though vertical scrollbar in components is still not fixed which is quite an inconvenience.

**Reply by ConorWebb96** [reply]
Hey @gmentzik,

This was completed and fixed as part of the last months iteration. https://github.com/Budibase/budibase/issues/10990

**Reply by gmentzik** [reply]
Thank you
